Source/items.cpp

```diff
@@ -3410,18 +3410,18 @@ void ProcessItems()
 		auto &item = Items[ii];
 		if (!item._iAnimFlag)
 			continue;
-		item.AnimInfo.ProcessAnimation();
+		item.AnimInfo.processAnimation();
 		if (item._iCurs == ICURS_MAGIC_ROCK) {
-			if (item._iSelFlag == 1 && item.AnimInfo.CurrentFrame == 10)
-				item.AnimInfo.CurrentFrame = 0;
-			if (item._iSelFlag == 2 && item.AnimInfo.CurrentFrame == 20)
-				item.AnimInfo.CurrentFrame = 10;
+			if (item._iSelFlag == 1 && item.AnimInfo.getCurrentFrame() == 10)
+				item.AnimInfo.setCurrentFrame(0);
+			if (item._iSelFlag == 2 && item.AnimInfo.getCurrentFrame() == 20)
+				item.AnimInfo.setCurrentFrame(10);

```

Comment:
I'm not sure this condition can ever be hit currently. Magic Rock has an animation length of 10. Starting with a currentFrame of 10 means it's immediately hitting the >= lastFrame condition in processAnimation and it gets reset to frame 0.